### PR TITLE
RAD-131 First Implementation of AcceptanceTests

### PR DESCRIPTION
**Pull request by manuelhintermayr**
see https://issues.openmrs.org/browse/RAD-131

### CODE REVIEW

**Comment by dkayiwa** [comment]
Can you squash the commits into one?

**Comment by manuelhintermayr** [comment]
done

**Comment by teleivo** [comment]
@manuelhintermayr please set your commit message title to the title of the pull request

**Comment by manuelhintermayr** [comment]
@teleivo done

File: acceptanceTest/build.gradle [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+
+        outputs.upToDateWhen { false }
+
+        systemProperty "geb.build.reportsDir", reporting.file("$name/geb")
+        systemProperty "geb.env", driver
+        if(System.getProperty("configFile")==null)
+        {
+            systemProperty "userName", "admin"
+            systemProperty "userPW", "Admin1234"
```
Comment (teleivo) [inline acceptanceTest/build.gradle]: please change to default openmrs admin pw = "Admin123". as this will just annoy most people ;)

Comment (teleivo) [inline acceptanceTest/build.gradle]: just ran the chromeTests and got following error

image

should this happen? Do the tests rely on data that has to be entered before running the tests?

Comment (manuelhintermayr) [inline acceptanceTest/build.gradle]: @teleivo Currently yes.
For this test scenario, the acceptanceTest picks two different orders: 
One with performed_status=="completed" and one with performed_status!="completed".
The driver clicks on those two orders and checks on the new opened page, if the performed_status is still the same.

Comment (teleivo) [inline acceptanceTest/build.gradle]: your test setup should then also make sure that the data needed by the tests is already in the database (as for ex. done by the ComponentTests you find in this module which have the data they rely on in dataSet.xml's which get imported before the tests run). it should be self-contained. the only thing it should rely on is that a server with db and openmrs is available on the specified endpoint.

Comment (manuelhintermayr) [inline acceptanceTest/build.gradle]: @teleivo  okay

"the only thing it should rely on is that a server with db and openmrs is available on the specified endpoint." 
- does this configuration normally also include the demo data for 5,000 sample patients from http://openmrs.org/download/ ?

Comment (teleivo) [inline acceptanceTest/build.gradle]: no. it includes for ex. https://github.com/openmrs/openmrs-module-radiologydcm4chee/blob/master/api/src/test/resources/org/openmrs/module/radiology/include/RadiologyServiceComponentTestDataset.xml or datasources already coming from the openmrs-core repository.

Comment (manuelhintermayr) [inline acceptanceTest/build.gradle]: Alright, I'm going to work on that

Comment (4ahif-hintermayr) [inline acceptanceTest/build.gradle]: @teleivo done

**Comment by teleivo** [comment]
@manuelhintermayr please remove all author attributions from the javadocs
see https://wiki.openmrs.org/display/docs/Coding+Conventions

File: acceptanceTest/readme.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
+# Description
+
+These are the AcceptanceTests for the OpenMRS Radiology Module.
+
+# Installation
+-) To use this gradle project you need to install Gradle
```
Comment (teleivo) [inline acceptanceTest/readme.md]: please improve your readme.
see https://help.github.com/articles/basic-writing-and-formatting-syntax/ for general formatting (your lists are off)
and https://help.github.com/articles/creating-and-highlighting-code-blocks/ for putting all your commands in code blocks

this improves readability :)
you should be able to edit your readme file on github, where you can directly see the result
or use githubs atom editor which has markdown rendering

File: acceptanceTest/src/test/groovy/org/openmrs/module/radiology/test/Configuration.groovy [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package org.openmrs.module.radiology.test
+
+/**
```
Comment (teleivo) [inline acceptanceTest/src/test/groovy/org/openmrs/module/radiology/test/Configuration.groovy]: remove

File: acceptanceTest/src/test/groovy/org/openmrs/module/radiology/test/pages/LoginPage.groovy [hunk 4]
```diff
@@ -0,0 +1,17 @@
+package org.openmrs.module.radiology.test.pages
+/**
```
Comment (teleivo) [inline acceptanceTest/src/test/groovy/org/openmrs/module/radiology/test/pages/LoginPage.groovy]: remove

File: acceptanceTest/readme.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+http://rominirani.com/2014/07/28/gradle-tutorial-part-1-installation-setup/
+-) Next, go into the "plugin"-folder, after that into the "mUpdate"-folder and run the following command:
+"gradle install"
+
```
Comment (teleivo) [inline acceptanceTest/readme.md]: replace above text, simply by

```
cd plugin/mUpdate
gradle install
```

**Pull request merged**